### DISCUSSION TITLE
Need help building an app for Windows that uses pydantic. It builds ok but returns ModuleNotFoundError.

### DISCUSSION THREAD

**Discussion by reriley** [flet-dev/flet#4035] [Q&A]
### Question

I'm a bit lost with debugging an app build for Windows. I have developed an app using pydantic (among other third-party non-pure Python packages). Using `flet build windows --verbose` it builds happily but on run it returns an error saying that the pydantic module is not found.

Does anyone have any idea what I'm doing wrong or what I should investigate?

### Code sample

```python
# main.py
import flet as ft
import app

def main(page: ft.Page):
    app.App(page)

if __name__ == '__main__':
    ft.app(target=main)

# app.py
import pathlib
from functools import partial

import flet as ft

from .state import GlobalState
from .router import Router, routes
from .io import write_from_global_state, read_to_global_state


class App:
   [more code]

# state.py (where the code fails)
import datetime
import pathlib
from typing import List, Optional, TypeVar, Generic, Union, Any, TYPE_CHECKING
import flet as ft
from uuid import UUID, uuid4
from abc import ABC, abstractmethod

from pydantic import BaseModel

[more code]

# requirements.txt
flet==0.24.1
pydantic==2.6.1
pydantic-core==2.16.2
pandas==2.2.0
pytest==8.0.0
pytest-cov
pytest-mock
openpyxl
pytest-snapshot
requests==2.31.0
PyYAML==6.0.1
python-dateutil==2.8.2
polars==0.20.31
PyPDFForm==1.4.29
Jinja2==3.1.3
weasyprint==62.3
```

### Error message

```bash
`Traceback (most recent call last):
  File "<string>", line 43, in <module>
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "[   ]Temp\serious_python_tempc5194498\main.py", line 4, in <module>
  File "[   ]Temp\serious_python_tempc5194498\app\__init__.py", line 1, in <module>
  File "[   ]Temp\serious_python_tempc5194498\app\app.py", line 10, in <module>
  File "[   ]Temp\serious_python_tempc5194498\app\state.py", line 9, in <module>
ModuleNotFoundError: No module named 'pydantic'
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by reriley** [comment]
Posting the build terminal output below:

> flet build windows --verbose
[19:54:07] Flutter bootstrap directory: C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL
           No additional Flutter dependencies!
[19:54:12] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.24.1 ✅                                                                                 
           Customized app icons and splash images ✅                                                                                                                                 
           Run subprocess: ['C:\\Users\\*****\\AppData\\local\\flutter\\bin\\dart.BAT', 'run', 'flutter_launcher_icons']
(   ●  ) Generating app icons(3.2s)
(  ●   ) Generating app icons ⏳... 
Got dependencies in `C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL`.
(   ●  ) Generating app icons ⏳... 
Built flutter_launcher_icons:flutter_launcher_icons.
(    ● ) Generating app icons ⏳...   ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.13.1)
  ════════════════════════════════════════════
(   ●  ) Generating app icons ⏳... • Creating default icons Android
( ●    ) Generating app icons ⏳... • Overwriting the default Android launcher icon with a new icon
(    ● ) Generating app icons ⏳... • Overwriting default iOS launcher icon with new icon
(     ●) Generating app icons ⏳... Creating Icons for Web...
( ●    ) Generating app icons ⏳... Creating Icons for Windows...
(     ●) Generating app icons ⏳... Creating Icons for MacOS...
(●     ) Generating app icons ⏳... 
✓ Successfully generated launcher icons
[19:54:25] Generated app icons ✅                                                                                                                                                    
           Run subprocess: ['C:\\Users\\*****\\AppData\\local\\flutter\\bin\\dart.BAT', 'run', 'serious_python:main', 'package',
           'C:\\Users\\*****\\PycharmProjects\\Form8621Calculator\\Form8621Calculator', '--dep-mappings', 'flet=flet-embed', '--req-deps', 'flet-embed', '--exclude', 'build']      
(     ●) Packaging Python app(1.4s)
Built serious_python:main.
(   ●  ) Packaging Python app ⏳... Running package command
Creating asset directory: C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\app
(  ●   ) Packaging Python app ⏳... Copying Python app from C:\Users\*****\PycharmProjects\Form8621Calculator\Form8621Calculator to C:\Users\*****\AppData\Local\Temp\serious_python_
tempc5194498
(   ●  ) Packaging Python app ⏳... Installing dependencies [flet-embed] with pip command to C:\Users\*****\AppData\Local\Temp\serious_python_tempc5194498\__pypackages__
Extracting Python distributive from C:\Users\*****\AppData\Local\Temp\cpython-3.11.6+20231002-x86_64-pc-windows-msvc-shared-install_only.tar.gz to C:\Users\*****\AppData\Local\Temp
\hostpython3.11_b0d7d4a9
(●     ) Packaging Python app ⏳... Compiling Python sources at C:\Users\*****\AppData\Local\Temp\serious_python_tempc5194498
( ●    ) Packaging Python app ⏳... Delete unnecessary files with extensions: [.py, .c, .h, .typed, .exe]
Delete unnecessary files and directories: [__pycache__, bin]
(   ●  ) Packaging Python app ⏳... Creating app archive at C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\app/app.zip from C:\Users\*****\AppData\Local\Temp\seriou
s_python_tempc5194498
( ●    ) Packaging Python app ⏳... Deleting temp directory C:\Users\*****\AppData\Local\Temp\serious_python_tempc5194498
(    ● ) Packaging Python app ⏳... Deleting Python directory C:\Users\*****\AppData\Local\Temp\hostpython3.11_b0d7d4a9
[19:55:46] Packaged Python app ✅                                                                                                                                                    
           Run subprocess: ['C:\\Users\\*****\\AppData\\local\\flutter\\bin\\flutter.BAT', 'build', 'windows']
(     ●) Building Windows app ⏳... (1.3s)
Downloading packages...
( ●    ) Building Windows app ⏳...   collection 1.18.0 (1.19.0 available)
  fl_chart 0.68.0 (0.69.0 available)
  flutter_launcher_icons 0.13.1 (0.14.0 available)
  flutter_lints 2.0.3 (4.0.0 available)
  http_parser 4.0.2 (4.1.0 available)
  js 0.6.7 (0.7.1 available)
  lints 2.1.1 (4.0.0 available)
  material_color_utilities 0.11.1 (0.12.0 available)
  meta 1.15.0 (1.16.0 available)
! package_info_plus 8.0.2 (overridden)
  screen_retriever 0.1.9 (0.2.0 available)
  sensors_plus 4.0.2 (6.0.1 available)
  sensors_plus_platform_interface 1.2.0 (2.0.0 available)
  toml 0.15.0 (0.16.0 available)
  url_strategy 0.2.0 (0.3.0 available)
! wakelock_plus 1.2.8 (overridden)
! web 1.0.0 (overridden)
  web_socket_channel 2.4.5 (3.0.1 available)
  window_manager 0.3.9 (0.4.2 available)
Got dependencies!
16 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
(    ● ) Building Windows app ⏳... 
(     ●) Building Windows app ⏳... windows/flutter/CMakeLists.txt does not use FLUTTER_TARGET_PLATFORM, updating.
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\Z
ERO_CHECK.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\serious_python_windows\CopyPythonDLLs.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\f
lutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\f
lutter\flutter_wrapper_app.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\f
lutter\flutter_wrapper_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\screen_retriever\screen_retriever_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\serious_python_windows\serious_python_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\format(42): warning STL4038: The contents of <format> are available only with C++20 or lat
er. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\plugins\serious_python_windows\serious_python_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\url_launcher_windows\url_launcher_windows_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\window_manager\window_manager_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\p
lugins\window_to_front\window_to_front_plugin.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\r
unner\form8621calculator.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\A
LL_BUILD.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(541,5): warning MSB8029: The Intermediate directory or Output directory
 cannot reside under the Temporary directory as it could lead to issues with incremental build. [C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\I
NSTALL.vcxproj]
(●     ) Building Windows a  387.3s
√ Built build\windows\x64\runner\Release\form8621calculator.exe
[20:02:21] Built Windows app ✅                                                                                                                                                      
           Copying build output from: C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL\build\windows\x64\runner\Release\*
[20:03:03] Copied build to build\windows directory ✅                                                                                                                                
           Deleting Flutter bootstrap directory C:\Users\*****\AppData\Local\Temp\flet_flutter_build_zgxye0ExDL
[20:03:06] Successfully built your Windows app! 🥳 Find it in  build\windows directory. 📁

**Comment by ap4499** [comment]
Hi - I think this problem looks a lot like the one that I have encountered with Pytorch.

We have found that Flet is removing some files when using Flet build, namely Bin, .py and .pyc. This causes the app to launch, but report that it is missing components, just like yours is. 

I have a built a runtime patch using files from cx_freeze. I have detailed the approach in this thread, and also shared some rough code on how the patching is working.

https://github.com/flet-dev/flet/issues/3060 
https://github.com/ap4499/PyTorchFlet_Patch

I'm unsure if you dev both Mac/Windows.. but if anything, I think the patch should be easier on Windows, as you should just be able to patch the Dir after Flet Build has created it, if not, then patching the runtime cache should be similar to the above. I am yet to do it on Windows as I used Flet Pack before realising I needed Flet Build for MacOS

**Reply by ap4499** [reply]
Also just seen this: https://github.com/flet-dev/flet/discussions/3141

Might be worth waiting.
